test/steps/http.go

```diff
@@ -56,31 +60,53 @@ func (data *Data) httpPostRequestOnServiceWithPathAndBody(serviceName, path stri
 }
 
 func (data *Data) httpPostRequestOnServiceIsSuccessfulWithinMinutesWithPathAndBody(serviceName string, timeoutInMin int, path string, body *messages.PickleStepArgument_PickleDocString) error {
-	framework.GetLogger(data.Namespace).Infof("httpPostRequestOnServiceWithPathAndBody with service %s, path %s, %s bodyContent %s and timeout %d", serviceName, path, body.GetMediaType(), body.GetContent(), timeoutInMin)
+	path = data.ResolveWithScenarioContext(path)
+	bodyContent := data.ResolveWithScenarioContext(body.GetContent())
+	framework.GetLogger(data.Namespace).Debugf("httpPostRequestOnServiceIsSuccessfulWithinMinutesWithPathAndBody with service %s, path %s, %s bodyContent %s and timeout %d", serviceName, path, body.GetMediaType(), bodyContent, timeoutInMin)
 	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
 	if err != nil {
 		return err
 	}
-	return framework.WaitForSuccessfulHTTPRequest(data.Namespace, "POST", routeURI, path, body.GetMediaType(), body.GetContent(), timeoutInMin)
+	return framework.WaitForSuccessfulHTTPRequest(data.Namespace, "POST", routeURI, path, body.GetMediaType(), bodyContent, timeoutInMin)
 }
 
 func (data *Data) httpGetRequestOnServiceWithPathShouldReturnAnArrayofSizeWithinMinutes(serviceName, path string, size, timeoutInMin int) error {
 	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
 	if err != nil {
 		return err
 	}
+
+	path = data.ResolveWithScenarioContext(path)
 	return framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("GET request on path %s to return array of size %d", path, size), timeoutInMin,
 		func() (bool, error) {
 			return framework.IsHTTPResponseArraySize(data.Namespace, "GET", routeURI, path, "", "", size)
 		})
 }
 
+func (data *Data) httpGetRequestOnServiceWithPathAndSetVariable(serviceName, path, variableName string) error {
+	path = data.ResolveWithScenarioContext(path)
+	framework.GetLogger(data.Namespace).Debugf("httpGetRequestOnServiceWithPathAndSetVariable with service %s, path %s", serviceName, path)
+	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
+	if err != nil {
+		return err
+	}
+
+	resultBody, err := framework.ExecuteHTTPRequestWithStringResponse(data.Namespace, "GET", routeURI, path, "", "")
+	if err != nil {
+		return err
+	}
+
+	data.ScenarioContext[variableName] = resultBody
+	return nil
+}
+
 func (data *Data) httpGetRequestOnServiceWithPathShouldContainAstringWithinMinutes(serviceName, path, responseContent string, timeoutInMin int) error {
 	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
 	if err != nil {
 		return err
 	}
 
+	path = data.ResolveWithScenarioContext(path)
```

Comment:
responseContent could be as well resolved or ? we may have placeholder IMO

Reply:
Added